lib/data/kinetic/Kinetic.js

```diff
@@ -0,0 +1,350 @@
+            this._sockets[index]._events.emit('deleteResponse', err);
+            break;
+        case kineticlib.ops.GET_RESPONSE:
+            this._sockets[index]._request = false;
+            if (err) {
+                this._sockets[index]._events.emit('getResponse', err);
+            } else {
+                this._startGet(index, pdu);
+            }
+            break;
+        default:
+            break;
+        }
+        return undefined;
+    }
+
+    _startGet(index, pdu) {
+        const chunk = this._sockets[index].sock.read();
+        this._sockets[index]._requestGet = true;
+        this._sockets[index]._chunkSize = pdu.getChunkSize();
+        this._count += chunk.length;
+        if (this._count === this._sockets[index]._chunkSize) {
+            const tag = crypto.createHmac('sha1', 'asdfasdf').update(chunk)
+                      .digest();
+            if (!pdu.checkTagIntegrity(tag)) {
+                this._sockets[index]._events.emit(
+                    'getResponse', errors.InternalError);
+            } else {
+                this._sockets[index]._events.emit('getResponse', null, chunk);
+            }
+        } else {
+            this._chunkTab.push(chunk);
+        }
+    }
+
+    _endGet(index) {
+        let chunk;
+        if (this._count !== this._sockets[index]._chunkSize) {
+            chunk = this._sockets[index].sock.read();
+            this._chunkTab.push(chunk);
+            this._count += chunk.length;
+        }
+        if (this._count === this._sockets[index]._chunkSize) {
+            this._sockets[index]._events.emit(
+                'getResponse', null, Buffer.concat(this._chunkTab));
+        }
+    }
+
+    _newSocket() {
+        const socket = new net.Socket().pause();
+        socket.setNoDelay();
+        socket.setKeepAlive(true);
+        socket.unref();
+        return socket;
+    }
+
+    _propError(pdu) {
+        const statusCode = pdu.getStatusCode();
+        if (statusCode === kineticlib.errors.NOT_FOUND) {
+            return errors.ObjNotFound;
+        }
+        if (statusCode !== kineticlib.errors.SUCCESS) {
+            return pdu.getErrorMessage();
+        }
+        return undefined;
+    }
+
+    /**
+     * set the sequence of the request.
+     * the sequence is a monotonically increasing number for each request in a
+     * TCP connection
+     *
+     * @param {number} index - the index of the socket.
+     * @param {number} sequence - the sequence to be set.
+     * @returns {Kinetic} - To allow for a functional style.
+     */
+    setSequence(index, sequence) {
+        if (sequence >= Number.MAX_VALUE) {
+            const temp = this._sockets[index].sock;
+            this._sockets[index].sock = this._newSocket();
+            this._sockets[index].sock.on('connect', () => {
+                this._bindReadable(this._sockets[index], index);
+            }).connect(this._sockets[index].drive);
+            temp.destroy();
```

Comment:
normally no, because we choose a socket with no current request we are just doing this to avoid an error with the maximum value of a number